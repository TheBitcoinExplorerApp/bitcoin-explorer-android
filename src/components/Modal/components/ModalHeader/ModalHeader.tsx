// eslint-disable-next-line import/no-extraneous-dependencies
import { View, Text, StyleSheet } from "react-native";
import CloseBtn from "../CloseBtn/CloseBtn";

type ModalHeaderProps = {
  handleModalClose: () => void;
  title: string;
};

const styles = StyleSheet.create({
  title: {
    color: "#fff",
    fontSize: 18,
  },
  headerModal: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
  },
});

export default function ModalHeader(props: ModalHeaderProps) {
  const { handleModalClose, title } = props;

  return (
    <View style={styles.headerModal}>
      <Text style={styles.title}>{title}</Text>
      <CloseBtn handleModalClose={handleModalClose} />
    </View>
  );
}
