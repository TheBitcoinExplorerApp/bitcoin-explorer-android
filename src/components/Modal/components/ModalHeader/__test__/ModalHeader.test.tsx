/* eslint-disable import/no-extraneous-dependencies */
import renderer from "react-test-renderer";
import ModalHeader from "../ModalHeader";

describe("ModalHeader", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<ModalHeader handleModalClose={() => {}} title="Title" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
