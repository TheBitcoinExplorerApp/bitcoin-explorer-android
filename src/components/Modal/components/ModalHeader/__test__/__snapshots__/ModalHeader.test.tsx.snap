// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalHeader renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 18,
      }
    }
  >
    Title
  </Text>
  <View
    accessibilityLabel="Close Modal"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#D9D9D9",
        "borderRadius": 100,
        "height": 36,
        "justifyContent": "center",
        "opacity": 1,
        "padding": 5,
        "position": "absolute",
        "right": 0,
        "width": 36,
      }
    }
  >
    <svg
      height={20}
      width={20}
    />
  </View>
</View>
`;
